test/test_web.py

```diff
@@ -9383,6 +9383,18 @@ def test_reset_settings(self):
         assert new_quiz['config']['completion_mode'] == quiz['completion_mode'], new_quiz
         assert new_quiz['config']['enabled'] == quiz['users'][0]['quiz']['config']['enabled'], new_quiz
 
+    @with_context
+    def test_reset_exception(self):
+        admin = UserFactory.create()
+        self.signin_user(admin)
+        quiz = {'enabled':True,'questions':10,'passing':5,'completion_mode':'short_circuit'}
+        project = ProjectFactory.create(owner=admin, info={'quiz':quiz})
+        TaskFactory.create_batch(20, project=project, n_answers=1, calibration=1)
+        assert admin.get_quiz_not_started(project)
+        quiz['users'] = [{ 'quiz': {'config': {'enabled': True, 'reset': True}}}]
```

Comment:
Why is it an error? No `id`?

Reply:
Currently the code expect that every user in users has id.

Reply:
https://github.com/bloomberg/pybossa/blob/master/pybossa/view/projects.py#L3224

I wanted to prove that the flash will be shown in case any exception happens. 

That use case of a user with no id should never happen.